### PR TITLE
feat(Collapse): Collapse support configure panel content by items 

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/pull/42553

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Collapse support configure panel content by items  |
| 🇨🇳 Chinese |  折叠面板支持通过 items 配置面板内容   |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 27e3857</samp>

This pull request refactors the Collapse component and its demos to use the new `items` prop instead of the deprecated `Panel` component. It also updates the type definitions, the documentation, and the `rc-collapse` dependency to support a new feature for the expand icon position.

### <samp>🤖 Generated by Copilot at 27e3857</samp>

*  Import the type definition of `CollapseProps` from the `rc-collapse` package in `Collapse.tsx` and the demo files, which is used to specify the props of the `Collapse` component. ([link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-3541251bb31318316b91e1f6710b646bd92edaff4318ca790c6401bea6962fc9R3), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-a2f709a1721eac15b6d70715382d8e3437c3c47fbc24d91970643602a987681bL1-R4), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-e68c02a5815e5a9c6a7851ad90c20cd1d4f4103e3c132d156bc77afea77fee2bL1-R4), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-833be645f48f81985c9a8c7aec835b5c7a9bdde11295e31450b42bc3f541a1a1L1-R4), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-d8ede31f882f5473fab952ec6f7bc407690b8683099d02de7e6fdabf976401bdL1-R4), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-d8cf3c5489f44133d1e8c087f5e923cb1cc93aa64ef2d53a1923a578f65d9c42L1-R6), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-2bf4f723b58175f0bfa95045cc92a868ef5195a840b86dbb342125f610978c17L1-R4), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-601a0d21ce6b45dc31970b5e2a346e6531918bb71ec634dc7e49ccf8a79cb472L1-R4), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-1058d770ba3b1820615351e9c61975d65401af0e8de615c1a90203767f02c907L1-R4), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-235242af85a91b896ab25b465c57a45d895a7ea2a4ba33822742789b430c0aecL1-R3))
* Add the `items` prop to the `ExpandIconPosition` type in `Collapse.tsx`, which is used to indicate the position of the expand icon in the `Collapse` component. ([link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-3541251bb31318316b91e1f6710b646bd92edaff4318ca790c6401bea6962fc9R24))
* Replace the use of `Panel` components with the `items` prop in the `Collapse` component in the demo files, which simplifies the code and avoids the deprecation warning of the `Panel` component. ([link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-a2f709a1721eac15b6d70715382d8e3437c3c47fbc24d91970643602a987681bL12-R30), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-e68c02a5815e5a9c6a7851ad90c20cd1d4f4103e3c132d156bc77afea77fee2bL17-R34), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-833be645f48f81985c9a8c7aec835b5c7a9bdde11295e31450b42bc3f541a1a1L13-R31), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-d8ede31f882f5473fab952ec6f7bc407690b8683099d02de7e6fdabf976401bdL12-R33), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-d8cf3c5489f44133d1e8c087f5e923cb1cc93aa64ef2d53a1923a578f65d9c42L29-R51), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-2bf4f723b58175f0bfa95045cc92a868ef5195a840b86dbb342125f610978c17L42-R65), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-601a0d21ce6b45dc31970b5e2a346e6531918bb71ec634dc7e49ccf8a79cb472L12-R30), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-1058d770ba3b1820615351e9c61975d65401af0e8de615c1a90203767f02c907L17-R42), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-235242af85a91b896ab25b465c57a45d895a7ea2a4ba33822742789b430c0aecL17-R32), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-e2edbfbc0753c99cabe338132ec93b4aa3eec72256672343b815e160a2bd1a2cL15-R25))
* Add a deprecation warning to the `CollapsePanel` component in `CollapsePanel.tsx`, which is no longer recommended to use since the `Collapse` component supports the `items` prop to configure the panels. ([link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-7b26d89bc5d6c4b0c3933b263e17bb939b25f88b0e51e76014a3a29ae22ca223R24-R26))
* Move the import of `ConfigContext` from `CollapsePanel.tsx` to `Collapse.tsx`, since the latter uses the `ConfigContext` to get the `prefixCls` and `direction` values. ([link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-7b26d89bc5d6c4b0c3933b263e17bb939b25f88b0e51e76014a3a29ae22ca223L4-R5))
* Update the version of the `rc-collapse` dependency from 3.5.2 to 3.6.0 in `package.json`, which supports the `items` prop in the `Collapse` component. ([link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L126-R126))
* Add the documentation of the `items` prop in the `Collapse` component, and links to the `ItemType` interface from the `rc-collapse` package in `index.en-US.md` and `index.zh-CN.md`. It also adds an alert message to inform the users to use the `items` prop instead of the `Panel` component. ([link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-80196f1c3a222b13f4be47d5ed6461a8767e35482104430410405c1bc047af75L47-R52), [link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-d54c8a5b222ce1fa0e8d1043fc79b037bc9eb27d38bd30409d690c8275ac2c62L48-R53))
* Remove the unused import of `Panel` from the `size.tsx` demo file, since the `Panel` component is no longer used. ([link](https://github.com/ant-design/ant-design/pull/42545/files?diff=unified&w=0#diff-e2edbfbc0753c99cabe338132ec93b4aa3eec72256672343b815e160a2bd1a2cL1-R3))

### CODE REVIEW

File: components/collapse/index.zh-CN.md [hunk 1]
```diff
@@ -45,9 +45,12 @@ coverDark: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*sir-TK0HkWcAAA
 | ghost | 使折叠面板透明且无边框 | boolean | false | 4.4.0 |
 | size | 设置折叠面板大小 | `large` \| `middle` \| `small` | `middle` | 5.2.0 |
 | onChange | 切换面板的回调 | function | - |  |
+| items | 折叠项目内容 | [ItemType](https://github.com/react-component/collapse/blob/27250ca5415faab16db412b9bff2c131bb4f32fc/src/interface.ts#L6) | - | 5.6.0 |
 
 ### Collapse.Panel
 
+<Alert message="&gt; 5.6.0 请使用 items 方式配置面板."></Alert>
```
Comment (MadCcc) [inline components/collapse/index.zh-CN.md:52]:
```suggestion
<Alert message="&gt;= 5.6.0 请使用 items 方式配置面板."></Alert>
```

File: components/collapse/Collapse.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import RightOutlined from '@ant-design/icons/RightOutlined';
 import classNames from 'classnames';
+import type { CollapseProps as OriCollapseProps } from 'rc-collapse';
```
Comment (MadCcc) [inline components/collapse/Collapse.tsx:3]:
```suggestion
import type { CollapseProps as RcCollapseProps } from 'rc-collapse';
```

File: components/collapse/CollapsePanel.tsx [hunk 3]
```diff
@@ -21,6 +21,9 @@ export interface CollapsePanelProps {
   collapsible?: CollapsibleType;
   children?: React.ReactNode;
 }
+/**
+ * @deprecated use `items` instead
+ */
```
Comment (MadCcc) [inline components/collapse/CollapsePanel.tsx:26]: 这个应该在 Collapse 的 type 上加好点？

File: components/collapse/demo/collapsible.tsx [hunk 4]
```diff
@@ -1,32 +1,36 @@
...
-
-const { Panel } = Collapse;
 
 const text = `
   A dog is a type of domesticated animal.
   Known for its loyalty and faithfulness,
   it can be found as a welcome guest in many households across the world.
 `;
 
+const items: CollapseProps['items'] = [
```
Comment (zombieJ) [inline components/collapse/demo/collapsible.tsx:11]: 这个不等价，原来的例子是多个 Collapse 不是一个

File: components/collapse/demo/borderless.tsx [hunk 5]
```diff
@@ -10,18 +9,24 @@ const text = (
...
-    <Panel header="This is panel header 3" key="3">
-      {text}
-    </Panel>
-  </Collapse>
-);
+const items: CollapseProps['items'] = [
+  {
+    key: '1',
+    label: 'This is panel header 1',
+    children: <p>{text}</p>,
```
Comment (zombieJ) [inline components/collapse/demo/borderless.tsx:16]: 和原来的测试不等价

File: components/collapse/__tests__/__snapshots__/demo.test.ts.snap [hunk 6]
```diff
@@ -319,11 +319,13 @@ exports[`renders components/collapse/demo/borderless.tsx correctly 1`] = `
       <div
         class="ant-collapse-content-box"
       >
+        <p />
```
Comment (zombieJ) [inline components/collapse/__tests__/__snapshots__/demo.test.ts.snap:322]: 这边多了空的应该是有问题的，需要看一下

File: components/collapse/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 7]
```diff
@@ -1420,7 +1420,7 @@ exports[`renders components/collapse/demo/noarrow.tsx extend context correctly 1
     </div>
   </div>
   <div
-    class="ant-collapse-item ant-collapse-no-arrow"
+    class="ant-collapse-item"
```
Comment (MadCcc) [inline components/collapse/__tests__/__snapshots__/demo-extend.test.ts.snap:1423]: 这里少 class 了

Comment (kiner-tang) [reply components/collapse/__tests__/__snapshots__/demo-extend.test.ts.snap:1423]: 我们现在已经不用 `panel` 了，而是用 `items` 配置了，这个 class 是用 `panel` 的时候才会有的
image

**Review by MadCcc — APPROVED** [review]

**Pull request merged**